marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/map-multiple-entities/createRelatedEntityMapping.sjs

```diff
@@ -0,0 +1,186 @@
+               <m:val>Name/LastName</m:val>
+            </lastName>
+         </m:optional>
+      </Customer>
+   </m:entity>
+   <m:entity name="mapping1-Product">
+      <Product xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+         <m:optional>
+            <productId xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:integer">
+               <m:val>ProductId</m:val>
+            </productId>
+         </m:optional>
+         <m:optional>
+            <productName xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:string">
+               <m:val>Name</m:val>
+            </productName>
+         </m:optional>
+      </Product>
+   </m:entity>
+   <m:entity name="mapping2-Order">
+      <Order xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+         <m:optional>
+            <orderId xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:integer">
+               <m:val>OrderId</m:val>
+            </orderId>
+         </m:optional>
+         <m:optional>
+            <orderedBy xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:integer">
+               <m:val>../CustomerID</m:val>
+            </orderedBy>
+         </m:optional>
+         <m:optional>
+            <deliveredTo xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:integer">
+               <m:val>SendToID</m:val>
+            </deliveredTo>
+         </m:optional>
+         <m:for-each>
+            <m:select>Products</m:select>
+            <lineItems datatype="array">
+               <m:call-template name="mapping2-Order.lineItems" />
+            </lineItems>
+         </m:for-each>
+      </Order>
+   </m:entity>
+   <m:entity name="mapping2-Order.lineItems">
+      <LineItem xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+         <m:optional>
+            <quantity xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:integer">
+               <m:val>Quantity</m:val>
+            </quantity>
+         </m:optional>
+         <m:optional>
+            <orderIncludes xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:integer">
+               <m:val>ProductId</m:val>
+            </orderIncludes>
+         </m:optional>
+      </LineItem>
+   </m:entity>
+   <m:output>
+      <entityInstance0>
+         <m:for-each>
+            <m:select>/</m:select>
+            <m:call-template name="mapping0-Customer" />
+         </m:for-each>
+      </entityInstance0>
+      <entityInstance1>
+         <m:for-each>
+            <m:select>/Orders/Products</m:select>
+            <m:call-template name="mapping1-Product" />
+         </m:for-each>
+      </entityInstance1>
+      <entityInstance2>
+         <m:for-each>
+            <m:select>/Orders</m:select>
+            <m:call-template name="mapping2-Order" />
+         </m:for-each>
+      </entityInstance2>
+   </m:output>
+</m:mapping>
+
+`);
+
+template = constructRelatedEntityMapping();
+assertions.push(
```

Comment:
I recommend adding more assertions to explain why certain lines are what they are. Doing a full-XML comparison is useful to verify correctness, but it doesn't provide any explanation as to why it's correct. I think assertions on each entity name would be useful to explain why they are what they are.